src/main/java/com/deviceinsight/kafka/health/KafkaConsumingHealthIndicator.java

```diff
@@ -54,7 +58,10 @@
 	private final long subscriptionTimeoutMs;
 
 	private final ExecutorService executor;
+	private final AtomicBoolean running;
+	private final CacheService<String> cacheService;
 
+	private KafkaCommunicationResult kafkaCommunicationResult;
```

Comment:
Since `running` uses `AtomicBoolean`, shouldn't `AtomicReference<KafkaCommunicationResult>` be used here? This field is accessed and written by multiple threads.